src/libraries/System.Net.Security/src/System/Net/NegotiateAuthenticationPal.Unix.cs

```diff
@@ -0,0 +1,803 @@
+
+                contextFlags |= clientOptions.AllowedImpersonationLevel switch
+                {
+                    TokenImpersonationLevel.Identification => Interop.NetSecurityNative.GssFlags.GSS_C_IDENTIFY_FLAG,
+                    TokenImpersonationLevel.Delegation => Interop.NetSecurityNative.GssFlags.GSS_C_DELEG_FLAG,
+                    _ => 0
+                };
+
+                _isServer = false;
+                _spn = clientOptions.TargetName;
+                _securityContext = null;
+                _requestedContextFlags = contextFlags;
+                _package = clientOptions.Package;
+                _channelBinding = clientOptions.Binding;
+                _packageType = GetPackageType(_package);
+
+                if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(this, $"Peer SPN-> '{_spn}'");
+
+                if (clientOptions.Credential == CredentialCache.DefaultCredentials ||
+                    string.IsNullOrWhiteSpace(clientOptions.Credential.UserName) ||
+                    string.IsNullOrWhiteSpace(clientOptions.Credential.Password))
+                {
+                    if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(this, "using DefaultCredentials");
+                    _credentialsHandle = AcquireDefaultCredential();
+
+                    if (_packageType == Interop.NetSecurityNative.PackageType.NTLM)
+                    {
+                        // NTLM authentication is not possible with default credentials which are no-op
+                        throw new PlatformNotSupportedException(SR.net_ntlm_not_possible_default_cred);
+                    }
+                    if (string.IsNullOrEmpty(_spn))
+                    {
+                        throw new PlatformNotSupportedException(SR.net_nego_not_supported_empty_target_with_defaultcreds);
+                    }
+                }
+                else
+                {
+                    _credentialsHandle = AcquireCredentialsHandle(clientOptions.Credential);
+                }
+            }
+
+            public UnixNegotiateAuthenticationPal(NegotiateAuthenticationServerOptions serverOptions)
+            {
+                Interop.NetSecurityNative.GssFlags contextFlags = serverOptions.RequiredProtectionLevel switch
+                {
+                    ProtectionLevel.Sign => Interop.NetSecurityNative.GssFlags.GSS_C_INTEG_FLAG,
+                    ProtectionLevel.EncryptAndSign => Interop.NetSecurityNative.GssFlags.GSS_C_INTEG_FLAG | Interop.NetSecurityNative.GssFlags.GSS_C_CONF_FLAG,
+                    _ => 0
+                };
+
+                // NOTE: Historically serverOptions.Policy was ignore on Unix without an exception
+                // or error message. We continue to do so for compatibity reasons and because there
+                // are no direct equivalents in GSSAPI.
```

Comment:
```suggestion
                // NOTE: Historically serverOptions.Policy was ignored on Unix without an exception
                // or error message. We continue to do so for compatibility reasons and because there
                // are no direct equivalents in GSSAPI.
```